main.js

```diff
@@ -0,0 +1,94 @@
+// Clase Producto
+class Producto {
+  constructor(nombre, precio) {
+    this.nombre = nombre;
+    this.precio = precio;
+  }
+
+  calcularSubtotal(cantidad) {
+    return this.precio * cantidad;
+  }
+}
+
+// Clase Carrito
+class Carrito {
+  constructor() {
+    this.productos = [];
+  }
+
+  agregarProducto(producto, cantidad) {
+    const subtotal = producto.calcularSubtotal(cantidad);
+    this.productos.push({
+      nombre: producto.nombre,
+      precio: producto.precio,
+      cantidad,
+      subtotal
+    });
+  }
+
+  calcularTotal() {
+    return this.productos.reduce((acc, item) => acc + item.subtotal, 0);
+  }
+
+  mostrarResumen() {
+    let mensaje = "🛒 RESUMEN DEL PEDIDO 🛒\n\n";
+    this.productos.forEach(p => {
+      mensaje += `${p.nombre} x${p.cantidad} = $${p.subtotal}\n`;
+    });
+    mensaje += `\nTOTAL: $${this.calcularTotal()}`;
+    alert(mensaje);
+  }
+}
+
+// Lista de productos disponibles
+const listaProductos = [
+  new Producto("Alimento para perro", 3500),
+  new Producto("Juguete de goma", 1200),
+  new Producto("Rascador para gatos", 4200),
+  new Producto("Collar con luz", 2800)
+];
+
+// Mostrar productos al usuario
+function mostrarCatalogo() {
+  let mensaje = "Productos disponibles:\n";
+  listaProductos.forEach((p, i) => {
+    mensaje += `${i + 1}. ${p.nombre} - $${p.precio}\n`;
+  });
+  alert(mensaje);
+}
+
+// Simulador de compra
+function simuladorDeCompra() {
+  const carrito = new Carrito();
+  let seguir = true;
+
+  alert("Bienvenido al simulador de PetShop Online 🐾");
+
+  while (seguir) {
+    mostrarCatalogo();
+    const opcion = parseInt(prompt("Ingresá el número del producto que querés agregar:"));
```

Comment:
¿Por qué no haces másseparada la lógica de la presentación? Pensa en más funciones, que despues puedas reemplazar facilmente. Por ejemplo:
```suggestion
function solicitarProductoIndex(productos) {
  const input = prompt("Ingresá el número del producto que querés agregar:");
  if (input === null) return null; // usuario canceló
  const index = parseInt(input);
  if (isNaN(index) || index < 1 || index > productos.length) return -1;
  return index - 1;
}
```